src/main/java/picard/vcf/LiftoverVcf.java

```diff
@@ -313,21 +323,155 @@ public static void main(final String[] args) {
+        if (!source.isBiallelic()) return null;  //only supporting biallelic indels, for now.
+
+        final Locatable target = getFlippedLocation(source, liftOver);
+        if (target == null) return null;
+
+        final Map<Allele, Allele> reverseComplementAlleleMap = new HashMap<>(10);
+
+        reverseComplementAlleleMap.clear();
+        final List<Allele> alleles = new ArrayList<>();
+
+        for (final Allele oldAllele : source.getAlleles()) {
+            // target.getStart is 1-based, reference bases are 0-based
+            final String alleleWithoutHead = oldAllele.getBaseString().substring(1, oldAllele.length());
+            final char refBase = (char) referenceSequence.getBases()[target.getStart() - 1];
+            final Allele fixedAllele = Allele.create(refBase + SequenceUtil.reverseComplement(alleleWithoutHead), oldAllele.isReference());
+            alleles.add(fixedAllele);
+            reverseComplementAlleleMap.put(oldAllele, fixedAllele);
+        }
+
+        VariantContextBuilder builder = new VariantContextBuilder(source.getSource(),
+                target.getContig(),
+                target.getStart(),
+                target.getEnd(),
+                alleles);
+
+        builder.id(source.getID());
+        builder.attributes(source.getAttributes());
+
+        builder.genotypes(fixGenotypes(source.getGenotypes(), reverseComplementAlleleMap));
+        builder.filters(source.getFilters());
+        builder.log10PError(source.getLog10PError());
+
+        return leftAlignVariant(builder.make(), referenceSequence);
+    }
+
+    protected static GenotypesContext fixGenotypes(final GenotypesContext originals, final Map<Allele, Allele> alleleMap) {
         // optimization: if nothing needs to be fixed then don't bother
-        if ( reverseComplementAlleleMap.isEmpty() ) {
+        if (alleleMap.isEmpty()) {
             return originals;
         }
 
         final GenotypesContext fixedGenotypes = GenotypesContext.create(originals.size());
-        for ( final Genotype genotype : originals ) {
-            final List<Allele> fixedAlleles = new ArrayList<Allele>();
-            for ( final Allele allele : genotype.getAlleles() ) {
-                final Allele fixedAllele = reverseComplementAlleleMap.containsKey(allele) ? reverseComplementAlleleMap.get(allele) : allele;
+        for (final Genotype genotype : originals) {
+            final List<Allele> fixedAlleles = new ArrayList<>();
+            for (final Allele allele : genotype.getAlleles()) {
+                final Allele fixedAllele = alleleMap.getOrDefault(allele, allele);
                 fixedAlleles.add(fixedAllele);
             }
             fixedGenotypes.add(new GenotypeBuilder(genotype).alleles(fixedAlleles).make());
         }
         return fixedGenotypes;
     }
-}
+
+    protected static VariantContext leftAlignVariant(final VariantContext vc, final ReferenceSequence referenceSequence) {
```

Comment:
Sounds reasonable for me, the same as I did with the `CreateSequenceDictionary` improvements... Thanks!